src/sql/workbench/services/objectExplorer/browser/objectExplorerService.ts

```diff
@@ -469,6 +459,43 @@ export class ObjectExplorerService implements IObjectExplorerService {
 		});
 	}
 
+	private completedExpandRequest(expandResult: NodeExpandInfoWithProviderId,
+		resultMap: Map<string, azdata.ObjectExplorerExpandInfo>,
+		node: TreeNode, newRequest: boolean,
+		session: azdata.ObjectExplorerSession,
+		allProviders: azdata.ObjectExplorerProviderBase[]): Promise<azdata.ObjectExplorerExpandInfo> {
+		return new Promise<azdata.ObjectExplorerExpandInfo>((resolve, reject) => {
```

Comment:
This doesn't need to create a new promise - there's no callbacks you have to call resolve from. And it also doesn't seem to need to return a Promise at all, I'm not seeing any async code in here. It can just return mergeResults directly.